File: php/classes/aplicacao/Local.php
<?php

class Local extends Aplicacao {
	
    protected $nome;
    protected $localizacao;
    protected $notas;
    
    public function Local($id = null) {
        if(!is_null($id)) {
            $locais = $this->consultar("id_local = {$id}");
            foreach($locais as $local) {
                $this->setNome($local->getNome());
                $this->setLocalizacao($local->getLocalizacao());
                $this->setNotas($local->getNotas());
                $this->setId($id);
            }
        }
    }
    
    public function inserir()
    {        
        if(strlen($this->nome) > 0)
            return Conexao::Inserir($this);
        throw new Exception("Local deve ter um nome.");
            
    }

    public function alterar()
    {
        if(strlen($this->nome) > 0)
            return Conexao::Editar($this);
        throw new Exception("Local deve ter um nome.");
    }

    public function excluir()
    {
        $valorId = $this->getId();
        if(is_null($valorId)) {
            throw new Exception("Local inexistente.");
        } else {    
            $evento = new Evento();
            $eventosAssociados = $evento->consultar("local = {$valorId}");
            if(count($eventosAssociados) == 0) {
                Conexao::Deletar($this);
                unset($this);
            } else {
                throw new Exception("Local tem eventos associados. Não é possível excluir.");
            }
        }             
    }

    public function consultar($consulta = null)
    {
        return Conexao::Listar($this, $consulta);
    }

}

File: java/classes/Local.java
package classes;

public class Local

{
	private String nome;
	private String localizacao;
	private String notas;

	public Local(String nome) {
		this.setNome(nome);
		this.setLocalizacao("");
		this.setNotas("");	
	}
	
	public Local(String nome, String localizacao, String notas) {
		this.setNome(nome);
		this.setLocalizacao(localizacao);
		this.setNotas(notas);
	}

	public void setNome(String nome) {
		this.nome = nome;
	}

	public String getNome() {
		return this.nome;
	}

	public void setLocalizacao(String localizacao) {
		this.localizacao = localizacao;
	}

	public String getLocalizacao() {
		return this.localizacao;
	}

	public void setNotas(String notas) {
		this.notas = notas;
	}

	public String getNotas() {
		return this.notas;
	}
}
